modules/data.remote/inst/SMAPvsModel_Comparison.R

```diff
@@ -0,0 +1,353 @@
+#' in a specified file directory for the Harvard Forest ('1000004945') site during the years 
+#' 2016 and 2017.
+#' 
+extractSoilMoist <- function(betySiteId, yearStart, yearEnd, ENS){
+
+  for (x in 0:(yearEnd - yearStart)) {
+    nc.file <- paste0(ensamble_dir, 'ENS-', ENS, '-', betySiteId, '/', yearStart + x,'.nc')
+    nc <- nc_open(nc.file)
+    if (x == 0) {
+      soilMoist <- ncvar_get(nc,'SoilMoist')
+      time <- PEcAn.utils::cf2datetime(nc[["dim"]][["time"]][["vals"]],nc$dim$time$units)
+    } else {
+      soilMoist <- append(soilMoist, ncvar_get(nc,'SoilMoist'))
+      time <- append(time, PEcAn.utils::cf2datetime(nc[["dim"]][["time"]][["vals"]],nc$dim$time$units))
+    }
+  }
+  return(list(time, soilMoist))
+}
+
+
+#' @title extractEnsamlbes
+#' @description utilizes extractSoilMoist function to extract an entire ensamble of a specified
+#' over the course of a specified date range.
+#'
+#' @param betySiteId bety site id for the desired location as (String)
+#' @param ENSnum number of members in the ensamble as (numeric)
+#'
+#' @return a data frame consisting of the soil mositure as well as their date-time values 
+#' during the specified date range
+#' 
+#'
+#' @examples 
+#' extractEnsambles(betySiteId = '1000004945', start = as.Date('2016-01-02'), end = as.Date('2016-07-01'))
+#' 
+#' Extracts all 25 ensample members (default) from the Harvard Forest (1000004945) site between the
+#' '2016-01-02' - '2016-07-01' date range.
+#' 
+extractEnsambles <- function(betySiteId, ENSnum=25){
+  for (x in 1:ENSnum) {
+    newEns <- extractSoilMoist(betySiteId = betySiteId, yearStart = as.numeric(format(start, '%Y')), 
+                               yearEnd = as.numeric(format(end, '%Y')), ENS = str_pad(x, 5, pad = '0'))
+    
+    if (x == 1){
+      moisture_time <- newEns
+    } else {
+      moisture_time[[1]] <- append(moisture_time[[1]], newEns[[1]])
+      moisture_time[[2]] <- append(moisture_time[[2]], newEns[[2]]) 
+    }
+  }
+  
+  # Eliminate extraneous date range values #
+  vtr <- as.data.frame(moisture_time, col.names = c('time', 'moisture'))
+  vtr <- vtr[vtr$time >= as.Date(start) & vtr$time <= as.Date(end), ]
+  
+  return(vtr)
+}
+
+
+#' @title runAnalysis
+#' 
+#' @description Runs multiple analyses on the specified site during a given date range
+#' and has the functionality to compare those values to actual soil moisture values
+#' collected from SMAP
+#'
+#' @param betySiteId bety site id for the desired location as (String)
+#' @param compare (boolean) to turn on/off comparison to actual smap data
+#'
+#' @return returns a list consisting of all of the analysis plots and statistics
+#'
+#' @examples 
+#' var <- runAnalysis('1000004945', '2016-01-02', '2016-07-16', TRUE)
+#' This runs an analysis on the Harvard Forest (1000004945) site with results between
+#' the date range of '2016-01-02' and '2016-07-16'
+#' 
+runAnalysis <- function(betySiteId, compare = TRUE){
```

Comment:
would benefit from a more specific name (especially as Analysis means something very specific in PEcAn's SDA module) e.g. runSMAPvalidation

Reply:
Renamed to runSMAPvalidation.